src/super_gradients/training/pipelines/pipelines.py

```diff
@@ -304,20 +313,25 @@ def _instantiate_image_prediction(self, image: np.ndarray, prediction: Detection
         return ImageDetectionPrediction(image=image, prediction=prediction, class_names=self.class_names)
 
     def _combine_image_prediction_to_images(
-        self, images_predictions: Iterable[ImageDetectionPrediction], n_images: Optional[int] = None
+        self, images_predictions: Iterable[ImageDetectionPrediction], n_images: Optional[int] = None, show_progress_bar: bool = True
     ) -> ImagesDetectionPrediction:
         if n_images is not None and n_images == 1:
             # Do not show tqdm progress bar if there is only one image
             images_predictions = [next(iter(images_predictions))]
         else:
-            images_predictions = [image_predictions for image_predictions in tqdm(images_predictions, total=n_images, desc="Predicting Images")]
+            images_predictions = [
+                image_predictions for image_predictions in tqdm(images_predictions, total=n_images, desc="Predicting Images", disable=not show_progress_bar)
+            ]
 
         return ImagesDetectionPrediction(_images_prediction_lst=images_predictions)
 
     def _combine_image_prediction_to_video(
-        self, images_predictions: Iterable[ImageDetectionPrediction], fps: float, n_images: Optional[int] = None
+        self, images_predictions: Iterable[ImageDetectionPrediction], fps: float, n_images: Optional[int] = None, show_progress_bar: bool = True
     ) -> VideoDetectionPrediction:
-        images_predictions = [image_predictions for image_predictions in tqdm(images_predictions, total=n_images, desc="Predicting Video")]
+        if not show_progress_bar:
```

Comment:
And this check does not make sense to me. Why are you changing logic of original implementation?